test/functional/mining.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test transaction selection (CreateNewBlock)."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.blocktools import *
+from test_framework.mininode import CTransaction, ToHex, FromHex, CBlock, CTxIn, CTxOut, COutPoint, CTxInWitness, COIN
+from test_framework.util import *
+from test_framework.script import *
+
+'''
+Test goals:
+    - Test that block max weight is respected (no blocks too big).
+    - Test that if the mempool has a tx that fits in a block, it will always be
+      included unless the block is within 4000 of max weight.
+    - Test that transaction selection takes into account feerate-with-ancestors
+      when creating blocks.
+    - Test that recently received transactions aren't selected unless the feerate
+      for including them is high.
+'''
+
+class MiningTest(BitcoinTestFramework):
+    def __init__(self):
+        self.num_nodes = 6
+        self.setup_clean_chain = True
+
+    def setup_network(self, split=False):
+        # Set blockmaxweight to be low, to require fewer transactions
```

Comment:
In commit "Add functional test for CreateNewBlock"

Could move comment down closer to -blockmaxweight setting.